src/Apps/Auction/routes.tsx

```diff
@@ -22,6 +23,51 @@ export const routes: RouteConfig[] = [
     path: "/auction-faq",
     Component: AuctionFAQ,
   },
+  {
+    path: "auction/:saleID/bid/:artworkID",
+    Component: BidRouteFragmentContainer,
+    render: ({ Component, props }) => {
+      console.log(props)
+      if (Component && props) {
+        const { artwork, me, location } = props as any
+        console.log({ artwork })
+        // TODO: Refine all logic here
+        if (!artwork) {
+          return <ErrorPage code={404} />
+        }
+
+        const redirect = findRedirectBid(artwork.sale_artwork.sale, me)
+        handleRedirect(redirect, location)
+
+        return <Component {...props} />
+      }
+    },
+    query: graphql`
+      query routes_BidQuery($saleID: String!, $artworkID: String!) {
+        artwork(id: $artworkID) {
+          ...Bid_artwork
```

Comment:
I would list all of these in-line w/o fragments, since currently the fragments aren't being split into their own components.